python/pyhail/keytable.py

```diff
@@ -121,12 +122,17 @@ def annotate(self, code, key_names=''):
 
         :param str code: Annotation expression.
 
-        :param str key_names: Comma separated list of field names to be treated as a key
+        :param key_names: field names to be treated as a key
+        :type key_names: str or list of str
 
         :rtype: :class:`.KeyTable`
         """
         try:
+            if isinstance(key_names, list):
+                key_names = ", ".join([str(l) for l in key_names])
```

Comment:
`key_names` is clearly documented as a list of str.  I think without the `str` is better.  I'd be inclined to check the type of the function at the top of the function, but this is not very "pythonic" since it goes against duck typing, e.g., you need an actual list instead of something behaves like a list (like some weirdo java proxy object), an actual string instead of something that behaves like a string, etc.  There is no way to check for list-like of string-like.  On the flip side, type errors often show up as long stack tracers deep inside libraries far from where the initial error is.

Even without it, you'll get a pretty reasonable error message:

```
>>> ','.join([1, 'foo', 13.4])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence item 0: expected string, int found
```

which is probably better than converting it into a string and getting an error from deeper inside Hail.